docs/src/main/tut/dsl.md

```diff
@@ -392,4 +392,48 @@ val averageTemperatureService = HttpService[IO] {
 }
 ```
 
+#### Optional query parameters
+
+To accept a optional query parameter a `OptionalQueryParamDecoderMatcher` can be used.
+
+```tut:book
+object OptionalYearQueryParamMatcher extends OptionalQueryParamDecoderMatcher[Year]("name")
+
+def getAverageTemperatureForCurrentYear(): IO[String] = ???
+def getAverageTemperatureForYear(y: Year): IO[String] = ???
+
+val service = HttpRoutes.of[IO] {
```

Comment:
In release-0.18.x, it's just `HttpService { ... }`.  In master, it's `HttpRoutes.of`.  That's why the build failed.  To get this into the 0.18 docs, change it to the former, and we can fix it up when it rolls forward to master.

Reply:
oh, thank you. Will do. I was pretty confused.